src/Databases/DatabaseOnDisk.cpp

```diff
@@ -751,8 +753,57 @@ ASTPtr DatabaseOnDisk::getCreateQueryFromStorage(const String & table_name, cons
     return create_table_query;
 }
 
-void DatabaseOnDisk::modifySettingsMetadata(const SettingsChanges & settings_changes, ContextPtr query_context)
+void DatabaseOnDisk::persistCreateDatabaseQuery(const ASTPtr & ast, const DatabaseMetadataPersistedCallback & commit_cb) const
+{
+    std::lock_guard lock(mutex);
+    persistCreateDatabaseQueryUnlocked(ast, commit_cb);
+}
+
+void DatabaseOnDisk::persistCreateDatabaseQueryUnlocked(const ASTPtr & ast, const DatabaseMetadataPersistedCallback & commit_cb) const
+{
+    auto database_metadata_path = getDatabaseMetadataFilePath();
```

Comment:
Looks like you lost fs::canonical for `context->getPath()`, previously it was done, now it is not. Previously there was:
```
fs::path metadata_root_path = fs::canonical(query_context->getGlobalContext()->getPath());
```
and now in `getDatabaseMetadataFilePath()` there is only `getPath()`.